extensions/ql-vscode/src/vscode-tests/no-workspace/remote-queries/repository-selection.test.ts

```diff
@@ -169,6 +176,101 @@ describe('repository-selection', function() {
         await expect(mod.getRepositorySelection()).to.be.rejectedWith(UserCancellationException, 'Invalid repository format');
       });
     });
+  });
+
+  describe('external repository lists file', async () => {
+    let pathExistsStub: sinon.SinonStub;
+    let fsStatStub: sinon.SinonStub;
+    let fsReadFileStub: sinon.SinonStub;
```

Comment:
Thank you so much for the write up! My eyes somehow completely glazed over the proxyrequire stuff 😬 

The explanation makes sense (and works! 🎉) - only thing I don't get is why it'd work if I ran the tests on their own. I'll go away and do some reading around it. Thanks again!

Reply:
I think it passed because somehow the tests were set up in a way that they would pass in a clean environment, but not one where other tests have already created state. I haven't looked at the details though.